packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -20,7 +21,7 @@ import 'shell.dart';
 
 abstract class AssetBundle {
   void close();
-  ImageResource loadImage(String key);
+  ImageResource loadImage(String key, [ AssetResolver resolver ]);
```

Comment:
I'm not sure whether the resolver should be passed in like this, or if it should be associated with the bundle in a more persistent way. The advantage of the latter is that it allows for additional caching, but the resolution strategy may vary by call site. One alternative would be to have different asset classes and have a resolution strategy per asset class.